File: find-the-duplicate-number/find-the-duplicate-number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        int result;
        for(int i=0; i<nums.size(); ++i){
            if(nums[abs(nums[i])] >= 0) nums[abs(nums[i])] = -nums[abs(nums[i])];
            else{
                result = abs(nums[i]);
                break;
            }
        }
        return result;
    }
};

File: find-the-duplicate-number/find-the-duplicate-number.java
class Solution {
    public int findDuplicate(int[] nums) {
        int first=0;
        while(nums[first] != nums[nums[first]]){
            int temp=nums[nums[first]];
            nums[nums[first]]=nums[first];
            nums[first]=temp;
        }
        return nums[first];
    }
}